DetectorDescription/DDCMS/plugins/DDHCalAngular.cc

```diff
@@ -0,0 +1,74 @@
+#include "DD4hep/DetFactoryHelper.h"
+#include "DetectorDescription/DDCMS/interface/DDPlugins.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//#define EDM_ML_DEBUG
+
+static long  algorithm(dd4hep::Detector& /* description */,
+                       cms::DDParsingContext& ctxt,
+                       xml_h e,
+                       dd4hep::SensitiveDetector& /* sens */) {
+
+  cms::DDNamespace      ns(ctxt,e,true);
+  cms::DDAlgoArguments  args(ctxt, e);
+  // Header section of original DDHCalAngular.h
+  int            n           = args.value<int>("n");
+  int            startCopyNo = args.value<int>("startCopyNo");
+  int            incrCopyNo  = args.value<int>("incrCopyNo");
+  double         rangeAngle  = args.value<double>("rangeAngle"); //Angular range
+  double         startAngle  = args.value<double>("startAngle"); //Start anle
+  double         shiftX      = args.value<double>("shiftX");     //x Shift
+  double         shiftY      = args.value<double>("shiftY");     //y Shift
+  double         zoffset     = args.value<double>("zoffset");    //z offset
+  dd4hep::Volume mother      = ns.volume(args.parentName());
+  dd4hep::Volume child       = ns.volume(args.value<std::string>("ChildName"));
+  // Increment
+  double         dphi        = rangeAngle/n;
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HCalGeom") << "DDHCalAngular: Parameters for positioning::"
+			       << " n " << n << " Start, Range, Delta " 
+			       << ConvertTo( startAngle, deg ) << " " 
+			       << ConvertTo( rangeAngle, deg ) << " " 
+			       << ConvertTo( dphi, deg )
+			       << " Shift " << shiftX << ":" << shiftY
+			       << "\n Parent " << mother.name() 
+			       << "\tChild " << child.name() 
+			       << " NameSpace " << ns.name();
+#endif
+  double theta  = 90._deg;
+  int    copy   = startCopyNo;
+  double phi    = startAngle;
+  for (int ii=0; ii<n; ++ii) {
+    double phix   = ConvertTo( phi, deg );
+    double phiy   = phix + 90._deg;
+    int    iphi   = (phix > 0) ? int(phix+0.1) : int(phix-0.1);
```

Comment:
Is rounding phix to the nearest degree actually necessary? I've seen and changed code like this previously, and my conclusion was that rounding to the nearest degree in general was not needed.

Reply:
Did that changes